src/Fantomas.Core/CodePrinter.fs

```diff
@@ -485,135 +460,24 @@ let genExpr (e: Expr) =
-                             +> match node.AnonRecord with
-                                | None ->
-                                    col sepNln node.Fields (fun e ->
-                                        // Add spaces to ensure the record field (incl trivia) starts at the right column.
-                                        addFixedSpaces targetColumn
-                                        // Lock the start of the record field, however keep potential indentations in relation to the opening curly brace
-                                        +> atCurrentColumn (genRecordFieldName e))
-                                | Some _ ->
-                                    col sepNln node.Fields (fun fieldNode ->
-                                        let genNode =
-                                            if ctx.Config.IndentSize < 3 then
-                                                sepSpaceOrDoubleIndentAndNlnIfExpressionExceedsPageWidth
-                                            else
-                                                sepSpaceOrIndentAndNlnIfExpressionExceedsPageWidth
-
-                                        // Add spaces to ensure the record field (incl trivia) starts at the right column.
-                                        addFixedSpaces targetColumn
-                                        +> atCurrentColumn (
-                                            enterNode fieldNode +> genIdentListNode fieldNode.FieldName
-                                        )
-                                        +> sepSpace
-                                        +> genSingleTextNode fieldNode.Equals
-                                        +> genNode (genExpr fieldNode.Expr)
-                                        +> leaveNode fieldNode)
-                             +> sepNlnWhenWriteBeforeNewlineNotEmpty
-                             +> (fun ctx ->
-                                 // Edge case scenario to make sure that the closing brace is not before the opening one
-                                 // See unit test "multiline string before closing brace"
-                                 let brace =
-                                     addFixedSpaces expressionStartColumn +> genSingleTextNode node.ClosingBrace
-
-                                 ifElseCtx lastWriteEventIsNewline brace (addSpaceIfSpaceAroundDelimiter +> brace) ctx))
-                            ctx
-
-            onlyIf isStruct !- "struct "
-            +> ifAlignOrStroustrupBrackets genMultilineRecordInstanceAlignBrackets genMultilineRecordInstance
+            genStructPrefix +> genMultilineRecord genCrampedFields node
 
-        fun ctx ->
-            let size = getRecordSize ctx node.Fields
-            genNode node.Node (isSmallExpression size smallRecordExpr multilineRecordExpr) ctx
+        genRecord smallRecordExpr multilineRecordExpr node
+    | Expr.InheritRecord node ->
+        let smallRecordExpr = genSmallInheritRecordExpr node
```

Comment:
`genSmallInheritRecordExpr` is only used here. Please move the definition up here.